front/app/modules/commercial/idea_custom_fields/admin/containers/projects/edit/ideaform/IdeaCustomField.tsx

```diff
@@ -234,101 +163,90 @@ export default memo<Props>(
     }, [fieldRequired, ideaCustomField, onChange]);
 
     const handleCollapseExpand = useCallback(() => {
-      onCollapseExpand(ideaCustomField.id);
+      onCollapseExpand(id);
       // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [ideaCustomField]);
+    }, [id]);
 
     if (!isNilOrError(ideaCustomField)) {
       return (
-        <Container className={`${className || ''} ${first ? 'first' : ''}`}>
-          <CollapsedContent
-            onMouseDown={removeFocusAfterMouseClick}
-            onClick={handleCollapseExpand}
-            className={`
-            ${collapsed ? 'collapsed' : 'expanded'}
-            e2e-${ideaCustomField.attributes.code}-setting-collapsed
-          `}
+        <Container className={`${className || ''}`}>
+          <StyledAccordion
+            title={
+              <HeadingContainer
+                onMouseDown={removeFocusAfterMouseClick}
+                className={`
+                e2e-${ideaCustomField.attributes.code}-setting-collapsed
+              `}
```

Comment:
Is there a related e2e test that needs to be updated as well?

Reply:
There is a related e2e test but it should still work with this.